packages/core/src/lib/core/queries/resolvers.ts

```diff
@@ -147,8 +148,8 @@ export async function findMany(
     })
   );
 
-  if (info.cacheControl && list.cacheHint) {
-    info.cacheControl.setCacheHint(
+  if (list.cacheHint) {
+    cacheControlFromInfo(info).setCacheHint(
```

Comment:
Could you use `maybeCacheControlFromInfo` instead of `cacheControlFromInfo` everywhere in Keystone internals since the schema will often be used outside of Apollo Server